pkg/apis/build/v1alpha1/lifecycle_image_validation.go

```diff
@@ -0,0 +1,44 @@
+package v1alpha1
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	LifecycleConfigName                          = "lifecycle-image"
+	LifecycleConfigNamespace                     = "kpack"
+	LifecycleConfigImageKey                      = "image"
+	LifecycleConfigPlatformApiVersionsAnnotation = "lifecycle.kpack.io/platformApiVersions"
+)
+
+type LifecycleImageValidator struct {
+	SupportedPlatformApiVersions []string
+}
+
+func (v LifecycleImageValidator) Validate(configMap *corev1.ConfigMap) (interface{}, error) {
```

Comment:
This is elegantly implemented but, seeing it makes me wonder if the validation is worth setting up a fringe webhook? At this time we don't expect any lifecycles to not be supported in the forseeable future. What does everyone think?

Reply:
I kinda agree. Is this worth it?

Reply:
I'm down to skip it for now

Reply:
considering that the current lifecycle image only supports 0.3+ for platform API, I would assume validation against the supported platform API does make sense. deprecation is bound to happen eventually (based on the precedence)

Reply:
The lifecycle started supporting multiple platform APIs at the 0.3 platform api and the team doesn't have plans to remove APIS in the near future.

Reply:
We can still validate the platform api @ builder creation.

Reply:
yes, validating at builder creation is possible, but takes the error further away from the cause (I think). In general, giving error feedback closer to the action causing might be more helpful - not a deal breaker.